### PR TITLE
Remove separate CI steps for sklearnex preview

### PR DESCRIPTION
**Pull request by Alexsandruss**
## Description

Separate testing for preview functionality of sklearnex is not required since there is no intersection between regular and preview scopes of sklearnex.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [ ] I have reviewed my changes thoroughly before submitting this pull request.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [ ] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [ ] I have added a respective label(s) to PR if I have a permission for that.
- [ ] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [ ] I have run it locally and tested the changes extensively.
- [ ] All CI jobs are green or I have provided justification why they aren't.
- [ ] I have extended testing suite if new functionality was introduced in this PR.

**Performance**

N/A

### CODE REVIEW

**Comment by icfaust** [comment]
This will complicate the work of daal4py deprecations, for example @richardnorth3 's work here uxlfoundation/oneDAL#3248 and uxlfoundation/oneDAL#3131, and any external contribution on uxlfoundation/oneDAL#3151 . I'm open for counter arguments, but i believe this can be closed until daal4py.sklearn is removed

**Comment by icfaust** [comment]
@Alexsandruss let us know how future daal4py deprecations should go with respect to benchmarks and then we can proceed here
